probe-rs/src/architecture/arm/core/exception_handling/armv6m.rs

```diff
@@ -41,21 +45,175 @@ impl From<u32> for ExceptionReason {
     }
 }
 
-pub fn exception_description(
-    _memory_interface: &mut dyn MemoryInterface,
-    stackframe_registers: &crate::debug::DebugRegisters,
-) -> Result<String, crate::Error> {
-    // Load the provided xPSR register as a bitfield.
-    let exception_number = Xpsr(
-        stackframe_registers
-            .get_register_value_by_role(&crate::core::RegisterRole::ProcessorStatus)?
-            as u32,
-    )
-    .exception_number();
-
-    // TODO: Some ARMv6-M cores (e.g. the Cortex-M0) do not have HFSR and CFGR registers, so we cannot
-    //       determine the cause of the hard fault. We should add a check for this, and return a more
-    //       helpful error message in this case (I'm not sure this is possible).
-    //       Until then, this will return a generic error message for all hard faults on this architecture.
-    Ok(format!("{:?}", ExceptionReason::from(exception_number)))
+impl ExceptionReason {
+    /// Expands the exception reason, by providing additional information about the exception from the
+    /// HFSR and CFSR registers.
+    pub(crate) fn expanded_description(
+        &self,
+        _memory: &mut dyn MemoryInterface,
+    ) -> Result<String, Error> {
+        match self {
+            ExceptionReason::ThreadMode => Ok("<No active exception>".to_string()),
+            ExceptionReason::Reset => Ok("Reset".to_string()),
+            ExceptionReason::NonMaskableInterrupt => Ok("NMI".to_string()),
+            ExceptionReason::HardFault => Ok("HardFault".to_string()),
+            ExceptionReason::SVCall => Ok("SVC".to_string()),
+            ExceptionReason::PendSV => Ok("PendSV".to_string()),
+            ExceptionReason::SysTick => Ok("SysTick".to_string()),
+            ExceptionReason::ExternalInterrupt(exti) => Ok(format!("External interrupt #{exti}")),
+            ExceptionReason::Reserved => {
+                Ok("<Reserved by the ISA, and not usable by software>".to_string())
+            }
+        }
+    }
+
+    /// Determines how the exception return address should be offset when unwinding the stack.
+    /// See Armv6-M Architecture Reference Manual, section B1.5.6.
+    pub(crate) fn is_precise_fault(
+        &self,
+        _memory: &mut dyn MemoryInterface,
+    ) -> Result<bool, Error> {
+        let is_precise = match self {
+            ExceptionReason::HardFault => {
+                // This should be true for syncrhonous exceptions, and false otherwise.
+                // TODO: Figure out how to differentiate that on ARMv6-M.
+                true
+            }
+            _ => false,
+        };
+        Ok(is_precise)
```

Comment:
Pretty sure clippy tells you to wrap the match in an Ok instead of defining a variable :)

Reply:
Clippy does not, but Noah does ... so consider it done ;-)

Reply:
I'm torn between two lovers ... who do I please? I'm sure there are Country & Western songs to help me decide, but in this case I will let you two duke it out the old fashioned way 😱🤣

Reply:
🤺

Reply:
Okay gotta admit this particuar case may be short enough :)

Reply:
For me it depends a bit on the case, but I really thought clippy would complain 🤔

Reply:
I'm going to leave it as per Noah's original suggestion.

It is the cases where the `Ok(long block of code)` contains `return Err` statements that usually makes me shy away from this syntax (just too confusing), but this case doesn't warrant too much analysis :)